app/components/Header.tsx

```diff
@@ -68,11 +68,25 @@ export function Header() {
   const email = useEmail()
   const [expanded, setExpanded] = useState(false)
   const [userMenuIsOpen, setUserMenuIsOpen] = useState(false)
-  const onClick = () => setExpanded((prvExpanded) => !prvExpanded)
+
+  function toggleMobileNav() {
+    setExpanded((expanded) => !expanded)
```

Comment:
just so I understand, why does does this have to be done this way instead of just !expanded?

Reply:
It probably doesn't make a difference here, but it makes a difference if you [call setState multiple times in the same evaluation of the functional component](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state).